article.md

```diff
@@ -0,0 +1,612 @@
+
+Let's start with the `Users` model. Firstly, we'll define an interface `UserAttributes`. This interface defines all the attributes you specify when creating a new instance of the model. In our database schema, we defined that a `User` has `id`, `name`, `createdAt` and `updatedAt` attributes. Let's see how that looks.
+
+```typescript
+// src/models/User.ts
+export interface UserAttributes {
+  id?: number;
+  name: string;
+  createdAt?: Date;
+  updatedAt?: Date;
+};
+```
+
+Simple. Note that `id`, `createdAt` and `updatedAt` are optional. This is because, when creating a new instance of a model, we don't want to be
+forced to specify these fields. Instead, we usually want Sequelize to automatically set them.
+
+The second type each model needs is a `ModelInstance`. This represents a Sequelize instance for an actual database row. For example, the function `User.find({ id: 1 })` returns an instance of the `User` model. It will have the `ModelAttributes` we defined (i.e., the table columns) as well as more Sequelize instance methods such as `.validate`, `.save`, `.update`, et cetera. For `Users`, the interface is defined as follows.
+
+```typescript
+// src/models/User.ts
+export interface UserInstance extends Sequelize.Instance<UserAttributes>, UserAttributes {
+  // At the moment, there's nothing more to add apart from the methods and attributes that the types `Sequelize.Instance<UserAttributes>` and `UserAttributes` give us. We'll add more here when we get on to adding associations.
+};
+```
+
+Now, we have the two types required. Next, we want to create a function that will actually define a Sequelize model for the `Users` table. The model should correspond to the `UserAttributes` and `UserInstance` types that we've already defined. We call this function `UserFactory`. In typical Sequelize manner, this function will take your Sequelize instance and a `DataTypes` object, and return the created Sequelize model. To define models in sequelize, we use the `sequelize.define<TInstance, TAttributes>()` function. The function takes an object which specifies the columns to create. 
+
+```typescript
+// src/models/User.ts
+export const UserFactory = (sequelize: Sequelize.Sequelize, DataTypes: Sequelize.DataTypes): Sequelize.Model<UserInstance, UserAttributes> => {
+  const attributes: SequelizeAttributes<UserAttributes> = {
+    name: {
+      type: DataTypes.STRING
+    }
+  };
+
+  const User = sequelize.define<UserInstance, UserAttributes>("User", attributes);
+
+  return User;
+};
+```
+
+Let's take a look at the `SequelizeAttributes<T>` type we use. To ensure that we pass `sequelize.define` an object that has definitions for all the attributes we specified in `UserAttributes`, we can create our own type `SequelizeAttributes<T>`. This is defined as follows. We'll put it in a `src/typings/` directory.
```

Comment:
maybe a note here that this _could_ be a global type - but that we could not get it to work with ts-node, hence manual import is required?

also, we kind of took this from that example repo, no? probably worth throwing in a reference.

Reply:
Added the reference.

I think that saying it could be a global type and explaining a possible bug in ts-node, is interesting. But, the article is already pretty long and I'm wary to add more content without it being directly relevant. Especially, as many readers may want to use manual import instead of global types anyway 😄